syntax = "proto3";

package stock;

import "proto/common.proto";

service StockService {
  rpc FindItem(ItemRequest) returns (Item);
  rpc AddStock(StockAdjustment) returns (common.Empty);
  rpc RemoveStock(StockAdjustment) returns (common.Empty);
}

message Item {
  string item_id = 1;
  int32 stock = 2;
  int32 price = 3;
}

message CreateItemRequest {
  int32 price = 1;
}

message CreateItemResponse {
  string item_id = 1;
}

message ItemRequest {
  string item_id = 1;
}

message StockAdjustment {
  string item_id = 1;
  int32 quantity = 2;
}
